CMakeLists.txt

```diff
@@ -284,25 +284,42 @@ endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 ## Find all dependencies
 
-##*****************************
-#find and include OpenMP if possible
+# Select shared memory parallelization
 if(${KRATOS_EXCLUDE_OPENMP} MATCHES ON)
-	message("**********************************************************")
-	message("OpenMP calculation was excluced artificially. Remove the var")
-	message("KRATOS_EXCLUDE_OPENMP from the configuration script or set it to off to include it")
-    message("**********************************************************")
-    if(${CMAKE_CXX_COMPILER_ID} MATCHES Intel)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -diag-disable 3180 ") # Disabling warning abt unrecognized omp pragmas
-    endif(${CMAKE_CXX_COMPILER_ID} MATCHES Intel)
-else(${KRATOS_EXCLUDE_OPENMP} MATCHES ON)
-	message(STATUS "OpenMP included")
+	message(FATAL_ERROR "The option \"KRATOS_EXCLUDE_OPENMP\" was removed, use KRATOS_SHARED_MEMORY_PARALLELIZATION=\"None\" instead")
+endif(${KRATOS_EXCLUDE_OPENMP} MATCHES ON)
+
+# default to OpenMP
+set(KRATOS_SHARED_MEMORY_PARALLELIZATION "OpenMP" CACHE STRING "Specify which thread library is used for shared-memory parallelization.")
```

Comment:
Maybe change this to something like:

```CMake
# Link on install
if(NOT DEFINED KRATOS_SHARED_MEMORY_PARALLELIZATION)
  SET(KRATOS_SHARED_MEMORY_PARALLELIZATION "OpenMP")
endif(NOT DEFINED KRATOS_SHARED_MEMORY_PARALLELIZATION)
```

Not really fan of the cache mechanism to handle undefined variables, we had problems in the past with that and nobody really uses CMake GUI for Kratos anyway

Reply:
done